### PR TITLE
Merge MichaelDinata-BugFix

### PR DESCRIPTION
**Pull request by michaeldinata**
Updated:
1. Using setters in parsers
2. Appendix D in DG

### CODE REVIEW

File: src/main/java/anichan/parser/AddToWatchlistParser.java [hunk 1]
```diff
@@ -62,7 +54,7 @@ private void parameterParser(String fieldGiven) throws AniException {
         isIntegerCheck(fieldValue, ANIME_ID);
 
         try {
-            addToWatchlistCommand.setAnimeIndex(Integer.parseInt(fieldValue));
+            return Integer.parseInt(fieldValue);
         } catch (NumberFormatException e) {
             throw new AniException(OUT_OF_BOUND_INDEX_ERROR);
         }
```
Comment (EyoWeiChin) [inline src/main/java/anichan/parser/AddToWatchlistParser.java]: Could this parseInt() be replaced with the parseStringToInteger method in CommandParser?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1211,4 +1211,68 @@ If you wish to add new checks, simply add the check file with a filename `check-
     4.  Other incorrect commands to try: 
         1.  `watchlist -d`
         2.  `watchlist -d x` (where x is a negative number, a word, or an additional parameter)
-        
\ No newline at end of file
+
```
Comment (EyoWeiChin) [inline docs/DeveloperGuide.md:1214]: Thanks for filling up the Manual testing section!

File: src/main/java/anichan/commands/ViewWatchlistCommand.java [hunk 3]
```diff
@@ -22,9 +22,25 @@
...
     protected static final String EMPTY_WATCHLIST_ERROR = "There are no anime in ";
     
-    private Integer watchlistIndex = 0;
+    private Integer watchlistIndex;
     private static final Logger LOGGER = AniLogger.getAniLogger(ViewWatchlistCommand.class.getName());
 
+    /**
+     * Creates a new instance of ViewWatchlistCommand.
+     */
+    public ViewWatchlistCommand() {
```
Comment (EyoWeiChin) [inline src/main/java/anichan/commands/ViewWatchlistCommand.java:31]: Is this constructor still needed?

File: src/main/java/anichan/parser/AddToWatchlistParser.java [hunk 4]
```diff
@@ -16,15 +16,6 @@
     protected static final String TOO_MUCH_ARGUMENTS = "Add To Watchlist command" + TOO_MUCH_FIELDS;
     protected static final String OUT_OF_BOUND_INDEX_ERROR = "Anime ID is invalid!";
```
Comment (EyoWeiChin) [inline src/main/java/anichan/parser/AddToWatchlistParser.java]: Since these constants are only used in this class, should this be private instead?

**Review by EyoWeiChin — CHANGES_REQUESTED** [review]
Thanks for making the changes! I left some comments that could help improve the overall code quality, hope it helps!

File: src/main/java/anichan/commands/ViewWatchlistCommand.java (hunk 3 above)
Comment (michaeldinata) [reply src/main/java/anichan/commands/ViewWatchlistCommand.java:31]: Yes, this constructor is needed as view command now takes in the optional parameter `-v <WATCHLIST_INDEX>`, and an empty constructor is required to not assign any value to the watchlist_index in viewCommand

File: src/main/java/anichan/parser/AddToWatchlistParser.java (hunk 4 above)
Comment (michaeldinata) [reply src/main/java/anichan/parser/AddToWatchlistParser.java]: Yes you are right, I will make amendments to that! :)

File: src/main/java/anichan/parser/AddToWatchlistParser.java (hunk 1 above)
Comment (michaeldinata) [reply src/main/java/anichan/parser/AddToWatchlistParser.java]: You're right. I missed that. Thanks for pointing it out!

File: src/main/java/anichan/commands/ViewWatchlistCommand.java (hunk 3 above)
Comment (EyoWeiChin) [reply src/main/java/anichan/commands/ViewWatchlistCommand.java:31]: Ah, I see, my mistake then.

**Review by EyoWeiChin — APPROVED** [review]
Thanks for making the changes, LGTM!!

**Review by OngDeZhi — APPROVED** [review]
LGTM :) Thanks for making the change so our codes are more consistent!

**Pull request merged**